### DISCUSSION TITLE
Virtual Threads

### DISCUSSION THREAD

**Discussion by mrguamos** [qos-ch/logback#714] [Q&A]
Since java 21 is out, do you have plans on replacing `ThreadLocal` with `ScopedValue` for virtual threads?

**Comment by sopgreg** [comment]
We would also welcome this change.

We have some callback classes which as injected as CDI beans and are shared across a common thread pool. When logging via `MDC`, the context is therefore leaking into other callback calls.

Our workaround is a `try-with-resources` like adapter which closes/clears the `MDC` when finishing the callback method. This would be much simpler and consise when using a `ScopedValue` for this purpose.
